qa/tasks/vstart_runner.py

```diff
@@ -719,27 +719,41 @@ def list_connections():
         pre_mount_conns = list_connections()
         log.info("Pre-mount connections: {0}".format(pre_mount_conns))
 
-        prefix = [os.path.join(BIN_PREFIX, "ceph-fuse")]
+        prefix = ['sudo', 'nsenter',
+                  '--net=/var/run/netns/{0}'.format(self.netns_name),
+                  os.path.join(BIN_PREFIX, "ceph-fuse")]
         if os.getuid() != 0:
             prefix += ["--client_die_on_failed_dentry_invalidate=false"]
-
         if mount_path is not None:
             prefix += ["--client_mountpoint={0}".format(mount_path)]
-
         if mount_fs_name is not None:
             prefix += ["--client_fs={0}".format(mount_fs_name)]
-
         prefix += mount_options;
-
-        self.fuse_daemon = self.client_remote.run(args=
-                                            ['nsenter',
-                                             '--net=/var/run/netns/{0}'.format(self.netns_name),
-                                            ] + prefix + [
-                                                "-f",
-                                                "--name",
-                                                "client.{0}".format(self.client_id),
-                                                self.mountpoint
-                                            ], wait=False)
+        fuse_cmd_args = prefix + ["-f", "--name",
+                                  "client.{0}".format(self.client_id),
+                                  self.mountpoint]
+        self.fuse_daemon = self.client_remote.run(args=fuse_cmd_args,
+                                                  wait=False)
+
+        # NOTE: ps -e | grep ceph-fuse just might fail in case ceph-fuse
+        # process wasn't lauched yet. in that case wait for few seconds and
+        # run the command again.
+        time.sleep(3)
+
+        # NOTE: when a command <args> is launched with sudo, 2 process are
+        # launched. one with sudo <args> and other with just <args>. Make sure
+        # we PID of latter one.
+        lines = self.client_remote.run(
+            args=['ps', '-ef', Raw('|'), 'grep', 'ceph-fuse'],
+            stdout=StringIO()).stdout.getvalue().strip().split('\n')
+        for line in lines:
+            if 'sudo' not in line and self.mountpoint in line and \
+               'client.' + self.client_id in line:
+                cols = line.split(' ')
+                self.fuse_daemon.subproc.pid = int([c for c in cols if c != ''][1])
+                break
+        else:
+            raise RuntimeError('this loop was supposed to get *the* PID')
```

Comment:
If the LocalRemoteProcess::kill() is not used, we can remove the kill() and then here will be the only user for subproc.pid. Then it will be simple and we can just set the asok_path to:
 
```
diff --git a/qa/tasks/vstart_runner.py b/qa/tasks/vstart_runner.py
index 6bde245..a27975a 100644
--- a/qa/tasks/vstart_runner.py
+++ b/qa/tasks/vstart_runner.py
@@ -585,7 +585,7 @@ class LocalKernelMount(KernelMount):
                 if asok_conf:
                     d = asok_conf.groups(1)[0]
                     break
-        path = "{0}/client.{1}.{2}.asok".format(d, self.client_id, self.fuse_daemon.subproc.pid)
+        path = "{0}/client.{1}.*.asok".format(d, self.client_id)
         log.info("I think my launching pid was {0}".format(self.fuse_daemon.subproc.pid))
         return path
```
Which will be like the one in: `qa/tasks/cephfs/fuse_mount.py`

```
401     def _asok_path(self):                                                                                                              
402         return "/var/run/ceph/ceph-client.{0}.*.asok".format(self.client_id)
403    
```

And then we can reuse the ` def admin_socket(self, args):` in fuse_mount.py with the following fix:
```

diff --git a/qa/tasks/cephfs/fuse_mount.py b/qa/tasks/cephfs/fuse_mount.py
index f46689b..b302cb0 100644
--- a/qa/tasks/cephfs/fuse_mount.py
+++ b/qa/tasks/cephfs/fuse_mount.py
@@ -415,6 +415,7 @@ import subprocess
 def find_socket(client_name):
         asok_path = "{asok_path}"
         files = glob.glob(asok_path)
+        mountpoint = "{mountpoint}"
 
         # Given a non-glob path, it better be there
         if "*" not in asok_path:
@@ -424,13 +425,16 @@ def find_socket(client_name):
         for f in files:
                 pid = re.match(".*\.(\d+)\.asok$", f).group(1)
                 if os.path.exists("/proc/{{0}}".format(pid)):
-                        return f
+                    with open("/proc/{0}/cmdline".format(pid), 'r') as f:
+                        if mountpoint in f.read():
+                            return f
         raise RuntimeError("Client socket {{0}} not found".format(client_name))
 
 print(find_socket("{client_name}"))
 """.format(
             asok_path=self._asok_path(),
-            client_name="client.{0}".format(self.client_id))
+            client_name="client.{0}".format(self.client_id),
+            mountpoint=self.mountpoint)
```

Reply:
> ```
> -        path = "{0}/client.{1}.{2}.asok".format(d, self.client_id, self.fuse_daemon.subproc.pid)
> +        path = "{0}/client.{1}.*.asok".format(d, self.client_id)
> ```

If I understand correctly, this won't work when we have multiple clients on the same node, right? By default vstart_runner.py creates at least 2 clients and the number might be greater some tests.

Reply:
It should work, since different clients have different different mountpoints, it will check the mountpoint from the `/proc/$pid/cmdline`.